sail/src/main/java/org/apache/rya/rdftriplestore/inference/InferenceEngine.java

```diff
@@ -467,126 +585,180 @@ private static void addParents(Vertex v, Set<URI> parents) {
+                }
+            }
+        }
+    }
+
+    protected void chainTransitiveProperty(final Resource subj, final URI prop, final Value obj, final Value core, final Set<Statement> sts, final boolean goUp, final Resource[] contxts) throws InferenceEngineException {
+        CloseableIteration<Statement, QueryEvaluationException> iter = null;
         try {
-            CloseableIteration<Statement, QueryEvaluationException> iter = RyaDAOHelper.query(ryaDAO, subj, prop, obj, conf, contxts);
+            iter = queryDao(subj, prop, obj, contxts);
             while (iter.hasNext()) {
-                Statement st = iter.next();
+                final Statement st = iter.next();
                 sts.add(new StatementImpl((goUp) ? (st.getSubject()) : (Resource) (core), prop, (!goUp) ? (st.getObject()) : (core)));
                 if (goUp) {
                     chainTransitiveProperty(null, prop, st.getSubject(), core, sts, goUp, contxts);
                 } else {
                     chainTransitiveProperty((Resource) st.getObject(), prop, null, core, sts, goUp, contxts);
                 }
             }
-            iter.close();
-        } catch (QueryEvaluationException e) {
+        } catch (final QueryEvaluationException e) {
             throw new InferenceEngineException(e);
+        } finally {
+            if (iter != null) {
+                try {
+                    iter.close();
+                } catch (final QueryEvaluationException e) {
+                    throw new InferenceEngineException("Error while closing \"chain transitive\" property statement iterator.", e);
+                }
+            }
         }
     }
 
     public boolean isInitialized() {
         return initialized;
     }
 
-    public void setInitialized(boolean initialized) {
+    public void setInitialized(final boolean initialized) {
         this.initialized = initialized;
     }
 
-    public RyaDAO getRyaDAO() {
+    public RyaDAO<?> getRyaDAO() {
         return ryaDAO;
     }
 
-    public void setRyaDAO(RyaDAO ryaDAO) {
+    public void setRyaDAO(final RyaDAO<?> ryaDAO) {
         this.ryaDAO = ryaDAO;
+        this.ryaDaoQueryWrapper = new RyaDaoQueryWrapper(ryaDAO);
     }
 
     public RdfCloudTripleStoreConfiguration getConf() {
         return conf;
     }
 
-    public void setConf(RdfCloudTripleStoreConfiguration conf) {
+    public void setConf(final RdfCloudTripleStoreConfiguration conf) {
         this.conf = conf;
     }
 
     public Graph getSubClassOfGraph() {
         return subClassOfGraph;
     }
 
+    /**
+     * Returns all types for which the specified type is considered a subclassOf
+     * based on the internal graph.
+     * @param type the type {@link URI} to find what it is a subclassOf.
+     * @return the {@link Set} of {@link URI} types that {@code type} is
+     * considered a subclassOf. Returns an empty set if nothing was found.
+     */
+    public Set<URI> getSubClassOfTypes(final URI type) {
```

Comment:
Yea, I was just trying to indicate that it was referencing the subClassOfGraph.  I renamed it to getSuperClasses to be less confusing.